tools/docker/Dockerfile

```diff
@@ -0,0 +1,231 @@
+		snapshotter/cmd/naive/naive_snapshotter \
+		/output
+
+
+
+
+#########################################
+#
+# VM IMAGES
+#
+#########################################
+
+
+
+# Build a rootfs for the microVM, including runc and firecracker-containerd's agent
+FROM alpine:3.8 as firecracker-vm-root
+COPY --from=runc-build /output/runc /usr/local/bin/
+COPY --from=firecracker-containerd-build /output/agent /usr/local/bin/
+ADD --chown=builder tools/docker/fc-agent.start /etc/local.d/fc-agent.start
+RUN apk add openrc \
+	&& ln -s /etc/init.d/local /etc/runlevels/default/local \
+	&& ln -s /etc/init.d/cgroups /etc/runlevels/default/cgroups \
+	&& ln -s /etc/init.d/agetty /etc/runlevels/default/agetty.ttyS0 \
+	&& ln -s /etc/init.d/devfs /etc/runlevels/boot/devfs \
+	&& ln -s /etc/init.d/hostname /etc/runlevels/boot/hostname \
+	&& ln -s /etc/init.d/procfs /etc/runlevels/boot/procfs \
+	&& ln -s /etc/init.d/sysfs /etc/runlevels/boot/sysfs
+
+# Convert the VM rootfs into an ext4 file. This step must run as root.
+FROM debian:stretch as firecracker-vm-root-builder
+COPY --from=firecracker-vm-root / /vm
+RUN mkdir -p /output \
+	&& cd /output \
+	&& mkfs.ext4 -d /vm vm.ext4 65536
+
+
+
+
+#########################################
+#
+# TEST IMAGES
+#
+#########################################
+
+
+
+# Just enough to run basic firecracker-containerd tests
+FROM debian:stretch as firecracker-containerd-test-base
+RUN --mount=type=cache,id=aptcache,target=/var/cache/apt \
+ 	--mount=type=cache,id=aptlib,target=/var/lib/apt,sharing=locked \
+	mkdir -p /etc/apt/sources.list.d \
+	&& echo "deb http://ftp.debian.org/debian stretch-backports main" > /etc/apt/sources.list.d/stretch-backports.list \
+	&& apt update \
+	&& apt --target-release stretch-backports install -y golang-go \
+	&& apt install -y \
+		build-essential \
+		git \
+		libdevmapper-dev \
+		libseccomp-dev \
+		ca-certificates
+
+COPY --from=firecracker-containerd-build /home/builder/firecracker-containerd firecracker-containerd
+RUN --mount=id=gomod,target=/tmp/go/pkg/mod,type=cache \
+	mkdir -p /root/go/pkg/mod \
+	&& cp -R /tmp/go/pkg/mod/* /root/go/pkg/mod
+ENTRYPOINT ["/bin/bash", "-c"]
+
+
+
+
+# Enough to start firecracker, firecracker-containerd and run full e2e tests. The entrypoint will start containerd
+# and the snapshotter in the background. The cmd drops to a shell if not otherwise overridden.
+FROM firecracker-containerd-test-base as firecracker-containerd-test-e2e
+COPY --from=firecracker-build /output/* /usr/local/bin/
+COPY --from=containerd-build /output/* /usr/local/bin/
+COPY --from=firecracker-vm-root-builder /output/vm.ext4 /var/lib/firecracker-containerd/runtime/hello-rootfs.ext4
+COPY --from=firecracker-containerd-build /output/* /usr/local/bin/
+COPY --from=runc-build /output/* /usr/local/bin/
+
+ADD https://s3.amazonaws.com/spec.ccfc.min/img/hello/kernel/hello-vmlinux.bin /var/lib/firecracker-containerd/runtime/hello-vmlinux.bin
```

Comment:
I'd prefer to see this as a `RUN` with `curl` and validating the downloaded file's sha256sum.